src/main/resources/META-INF/resources/primefaces/photocam/photocam.js

```diff
@@ -1163,6 +1202,33 @@ PrimeFaces.widget.PhotoCam = PrimeFaces.widget.BaseWidget.extend({
         } else {
             PrimeFaces.error('Capture error: AdvancedPhotoCam not attached to the camera');
         }
+    },
+    
+    /**
+     *  Retrieves the available video input device list 
+     *  @return {Promise<InputDeviceInfo[]>} Returns a promise to resolve the enumeration or null if the browser doesn't support media devices enumeration
```

Comment:
This is a little bit pedantic, but TypeScript types allow us to be explicit about the nullness of values (personally I prefer to use `undefined` in favor of `null`, but I don't think there's any consensus on that, so it's fine as it is):

```suggestion
     *  @return {Promise<InputDeviceInfo[]> | null} Returns a promise to resolve the enumeration or `null` if the browser does not support media devices enumeration.
```

But other than that, the JavaScript docs look pretty good : ) I'm glad it's being maintained by everybody now.